backend/src/personnel/personnel.service.ts

```diff
@@ -101,18 +106,55 @@ export class PersonnelService {
    * @param role 
    * @returns 
    */
-    async updateBcwsPersonnel(id: string,
+    async updateBcwsPersonnel(
+      id: string,
       personnel: UpdateBcwsPersonnelDTO,
-      role: Role){
-    
+      role: Role
+    ){
       this.logger.log(`Updating personnel ${id}`);
       const person = await this.personnelRepository.findOne({ where: { id } });
       const bcws = await this.bcwsPersonnelRepository.findOne({
         where: { personnel: { id } },
       });
   
-      this.logger.log(`${JSON.stringify(personnel)}`);
-  
+      // this.logger.log(`${JSON.stringify(personnel)}`);
+      if (personnel.tools && personnel.tools[0].hasOwnProperty('tool')) {
```

Comment:
Some really awkward stuff going on down here. Can be cleaned up by cleaning up dtos a little bit and/or gathering ids in a separate request from the frontend

Reply:
Languages is especially awkward, and a diagnosis on what's wrong with that entity would be tech debt

Reply:
![Screenshot 2024-05-27 at 10 13 56 AM](https://github.com/bcgov/talent-cloud/assets/78890675/fc728df0-b8a5-4ee5-befb-059be2a68591)
I think we should remove personnel_id and use personnel.personnel_id here, since that is the FK?

Reply:
(but also like you said can be done later as tech debt)

Reply:
personnel_id exists as a column anyway, its removal wouldn't serve a meaningful purpose, but having it allows us to use it as a relation for saving I think.